metricbeat/module/prometheus/remote_write/remote_write.go

```diff
@@ -62,6 +80,37 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	return m, nil
 }
 
+// MetricSetBuilder returns a builder function for a new Prometheus metricset using
+// the given namespace and event generator
+func MetricSetBuilder(namespace string, genFactory RemoteWriteEventsGeneratorFactory) func(base mb.BaseMetricSet) (mb.MetricSet, error) {
```

Comment:
Hmm, `genFactory` is not that static. It is different for the 2 different versions:
- `DefaultRemoteWriteEventsGeneratorFactory`
- `remoteWriteEventsGeneratorFactory`

Did you have anything in mind of how this could be handled with `New`  function?

Reply:
oh you are right! solving this, sorry for the noise